### PR TITLE
Create a FSharp.Tools NuGet package - closes #408

### LINKED ISSUE #408: Bundle the compiler and tools in a Nuget
As it recently came up, it would be often useful in various automation scenarios (Azure deployments, etc.) to fetch and use the compiler tools (`fsc.exe` and `fsi.exe`) without requiring a previous F# installation on the executing machine.

### PR DESCRIPTION
**Pull request by forki**
what else do we need?

/cc @dsyme @granicz

### CODE REVIEW

**Comment by dsyme** [comment]
Well, you need FSharp.Compiler.dll, FSharp.Compiler.Server.Shared.dll, FSharp.Compiler.InteractiveSettings.dll, fsiAnyCpu.exe. I also think it's correct to add FSharp.Build.dll and Microsoft.FSharp.Targets - in theory they feel as if they could be in a separate package, but I believe FSharp.Build.dll looks for fsc.exe in the same directory that it resides in by default, so it's appropriate to bundle them.

File: FSharp.Tools.Nuget/FSharp.Tools.3.1.nuspec [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+    <version>3.1.2.2</version>
+    <title>FSharp.Tools for F# 3.1</title>
+    <authors>F# Software Foundation</authors>
+    <owners>F# Software Foundation</owners>
+    <licenseUrl>https://github.com/fsharp/fsharp/blob/master/LICENSE</licenseUrl>
+    <projectUrl>https://github.com/fsharp/fsharp</projectUrl>
+    <iconUrl>http://fsharp.org/img/logo.png</iconUrl>
+    <requireLicenseAcceptance>false</requireLicenseAcceptance>
+    <description>
+      FSharp.Toos for F# 3.1
```
Comment (dsyme) [inline FSharp.Tools.Nuget/FSharp.Tools.3.1.nuspec]: Toes?  ToDos? TuTus?

Comment (forki) [inline FSharp.Tools.Nuget/FSharp.Tools.3.1.nuspec]: lol

File: FSharp.Tools.Nuget/FSharp.Tools.3.1.nuspec [hunk 2]
```diff
@@ -0,0 +1,34 @@
+<?xml version="1.0" encoding="utf-8"?>
+<package xmlns="http://schemas.microsoft.com/packaging/2010/07/nuspec.xsd">
+  <metadata>
+    <id>FSharp.Tools</id>
```
Comment (dsyme) [inline FSharp.Tools.Nuget/FSharp.Tools.3.1.nuspec]: Please change to FSharp.Compiler.Tools

**Comment by tpetricek** [comment]
I think this might still not work out-of-the box if you run this on a clean machine (i.e. Azure VM). 

At least I recently tried running `FSharp.Compiler.Service.dll` on an Azure VM with plain OS and it did not work - it turns out that `FSharp.Compiler.Service.dll` has dependencies on some MSBUILD stuff (`Microsoft.Build.dll` and a bunch of others) that are not available on clean install with .NET and have to be [installed separately](http://www.microsoft.com/en-gb/download/details.aspx?id=40760).

Not sure if we can get the MSBUILD stuff from NuGet as a dependency or something...?

**Comment by forki** [comment]
ok I added these. I wonder if we should depend on FSharp.Core package, but I assume since we are an app this is not really a good idea and we would need to copy files anyway,

**Comment by granicz** [comment]
Hmm, Tomas, do you remember what dependencies were those?

**Comment by dsyme** [comment]
Yes, for this I believe you should add all the relevant MSBuild DLLs as local copies.  I think you can just change [these references](https://github.com/fsharp/fsharp/blob/master/src/fsharp/FSharp.Compiler/FSharp.Compiler.fsproj#L512) to be `<Private>True</Private>` and add the DLLs from the directory.

**Comment by forki** [comment]
will look at this tomorrow. Now I have to watch Bayern vs. BVB ;-)

2015-04-28 20:06 GMT+02:00 Don Syme <notifications@github.com>:

**Comment by dsyme** [comment]
Yes, first things first @forki !

**Comment by forki** [comment]
done

**Comment by dsyme** [comment]
Could you change the name per [this comment](https://github.com/fsharp/fsharp/pull/409/files#r29268699)

File: FSharp.Tools.Nuget/FSharp.Tools.3.1.nuspec (hunk 2 above)
Comment (forki) [inline FSharp.Tools.Nuget/FSharp.Tools.3.1.nuspec]: This makes sense, but how about fsharp.compiler?

**Comment by forki** [comment]
done

File: src/fsharp/FSharp.Compiler/FSharp.Compiler.fsproj [hunk 3]
```diff
@@ -509,11 +509,19 @@
...
     <Reference Include="System.Runtime.Remoting" />
-    <Reference Include="Microsoft.Build.Framework, Version=12.0.0.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a" />
-    <Reference Include="Microsoft.Build.Engine, Version=12.0.0.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a" />
+    <Reference Include="Microsoft.Build.Framework, Version=12.0.0.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a">
+      <Private>True</Private>
+    </Reference>
+    <Reference Include="Microsoft.Build.Engine, Version=12.0.0.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a">
+      <Private>True</Private>
+    </Reference>
     <Reference Include="Microsoft.Build, Version=12.0.0.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a" />
```
Comment (dsyme) [inline src/fsharp/FSharp.Compiler/FSharp.Compiler.fsproj]: What about this one (Microsoft.Build?)

Comment (forki) [inline src/fsharp/FSharp.Compiler/FSharp.Compiler.fsproj]: done

**Pull request merged**